api/handlers.go

```diff
@@ -15,7 +15,7 @@ import (
 type GetVarsFunc func(r *http.Request) map[string]string
 
 // CreateSessionHandlerFunc returns a function that generates a session. Method = "POST"
-func CreateSessionHandlerFunc(sess Session, cache Cache) http.HandlerFunc {
+func CreateSessionHandlerFunc(updater Updater, cache Cache) http.HandlerFunc {
```

Comment:
This is clearer that it is an interface now, but it's lost the context that it relates to a session. Can I suggest…
```suggestion
func CreateSessionHandlerFunc(session SessionUpdater, cache Cache) http.HandlerFunc {
```
Along with all the other changes to make this possible.